### PR TITLE
Allow navigating to pack and other internal links directly from announcements

### PR DESCRIPTION
**Pull request by ClementPasteau**
## Pack not owned

https://github.com/4ian/GDevelop/assets/4895034/42ca0ec7-6931-473b-ae65-08dcb9c06ac6

## Pack owned 

https://github.com/4ian/GDevelop/assets/4895034/80d2f885-ac0a-4ac9-9ca8-6585d6ffac07

### CODE REVIEW

File: newIDE/app/src/AnnouncementsFeed/index.js [hunk 1]
```diff
@@ -75,23 +78,46 @@ export const AnnouncementsFeed = ({
...
+            const onInternalClick = () => navigateToRoute(route, otherParams);
+            const messageWithoutHref = message.replace(
+              markdownClickableImageRegex,
+              `![${groups.alt}](${groups.imageSource})`
+            );
+            return {
+              title,
+              message: messageWithoutHref,
+              isMarkdownImageOnly: true,
+              onInternalClick,
```
Comment (4ian) [inline newIDE/app/src/AnnouncementsFeed/index.js:106]: Could almost be called onClick?

Does not matter what this thing does, it just is called when there is a click

File: newIDE/app/src/AnnouncementsFeed/index.js [hunk 2]
```diff
@@ -65,6 +67,7 @@ export const AnnouncementsFeed = ({
     title: string,
     message: string,
     isMarkdownImageOnly?: boolean,
+    onInternalClick?: () => void,
   |} => {
```
Comment (4ian) [inline newIDE/app/src/AnnouncementsFeed/index.js:71]: I would almost move this into a standalone function outside of the component and unit test it, as the regex logic is growing a bit complex.

**Review by 4ian — APPROVED** [review]
While you're at it, any chance you can fix the wrong background color on the announcements?

image

We've been missing a Background or something there since ages

File: newIDE/app/src/AnnouncementsFeed/index.js (hunk 1 above)
Comment (ClementPasteau) [reply newIDE/app/src/AnnouncementsFeed/index.js:106]: yeah

File: newIDE/app/src/AnnouncementsFeed/index.js (hunk 2 above)
Comment (ClementPasteau) [reply newIDE/app/src/AnnouncementsFeed/index.js:71]: good idea

**Review by 4ian — APPROVED** [review]

**Pull request merged**